pkg/apis/app/v1beta1/types.go

```diff
@@ -58,6 +59,7 @@ type FlinkApplicationSpec struct {
 	AllowNonRestoredState          bool                `json:"allowNonRestoredState,omitempty"`
 	ForceRollback                  bool                `json:"forceRollback"`
 	MaxCheckpointRestoreAgeSeconds *int32              `json:"maxCheckpointRestoreAgeSeconds,omitempty"`
+	MaxCheckpointDeployAgeSeconds  *int32              `json:"maxCheckpointDeployAgeSeconds,omitempty"`
```

Comment:
Is `MaxCheckpointDeployAgeSeconds` functionally different from the `MaxCheckpointRestoreAgeSeconds`?

Reply:
Not functionally, no. my thinking was that these checkpoint age values could have very different acceptability thresholds since the `MaxCheckpointDeployAgeSeconds` would be looking for an ideal, possibly short window. I may have read the code wrong, but my thinking was by failing this age check, the operator would effectively poll the JM API looking for the next checkpoint to run and complete (we have ours checkpointing frequently.)

I may have read the state machine handler incorrectly though.